Homework 18.

IgushkinV opened the pull request:
ChangeLog вернул на начало 17 домашки.
Тесты пока не заработали.

IgushkinV committed:
Homework 18.

ChangeLog вернул на начало 17 домашки.
Тесты пока не заработали.

IgushkinV committed:
Homework 18.

Небольшой фикс сервисов.

tlt-java-courses commented on homework-api/pom.xml:
```diff
@@ -40,20 +45,11 @@
-            <artifactId>spring-webmvc</artifactId>
-        </dependency>
         <dependency>
             <groupId>javax.servlet</groupId>
```
нам тут точно все депенды нужны?

IgushkinV replied:
Убрал лишние.

tlt-java-courses commented on homework-impl/src/main/java/com/epam/igushkin/homework/Application.java:
```diff
@@ -1,44 +1,17 @@
-        dispatcher.addMapping("/*");
-
-        log.info("Application is running");
+    @Logging
```
этот метод у нас тоже залогируется?

IgushkinV replied:
Это был эксперимент. Забыл убрать.

tlt-java-courses commented on homework-impl/src/main/java/com/epam/igushkin/homework/config/WebMvcConfig.java:
```diff
@@ -1,5 +1,6 @@
 package com.epam.igushkin.homework.config;
```
нам точно нужен этот конфиг? напоминаю мы перешли на spring-boot

IgushkinV replied:
Да, удалил. Оставлял, пока переходил.

tlt-java-courses commented on homework-impl/src/main/java/com/epam/igushkin/homework/resources/impl/MyExceptionHandler.java:
```diff
@@ -13,8 +13,8 @@
 
     @ExceptionHandler(NoEntityFoundException.class)
     @ResponseStatus(value = HttpStatus.BAD_REQUEST, reason = "В базе не существует запись с указанным id.")
```
Обычно, когда мы ничего не нашли и не можем вернуть, то возвращаем 404 статус, 400 тут как-то не к месту.

IgushkinV replied:
Исправил.

tlt-java-courses commented on homework-impl/src/test/java/com/epam/igushkin/homework/resources/impl/CustomerResourceImplTest.java:
```diff
@@ -0,0 +1,23 @@
+
+@WebMvcTest(CustomerResource.class)
+@ExtendWith(SpringExtension.class)
+class CustomerResourceImplTest {
```
Тут что-то еще должно быть?

IgushkinV replied:
В процессе было. Теперь есть.

tlt-java-courses commented on homework-impl/src/test/java/com/epam/igushkin/homework/services/impl/CustomerServiceImplTest.java:
```diff
@@ -0,0 +1,37 @@
+                .setPhone("+7-5555-4444");
+        customerService.save(customer);
+        Customer savedCustomer = customerService.findById(1);
+                assertEquals(customer.getCustomerName(), savedCustomer.getCustomerName());
```
отступы

IgushkinV replied:
Аналогично.

IgushkinV committed:
Homework 18.

Добавил тесты сервисов.

IgushkinV committed:
Homework 18.

Правка теста.

IgushkinV committed:
Homework 18.

Исправил замечания.

Тесты ресурсов еще в процессе.

IgushkinV committed:
Homework 18.

Добавлены тесты на все методы класса CustomerResourceImpl.

IgushkinV committed:
Homework 18.

Добавлен тест на валидатор для CustomerServiceImpl.

Добавлены тесты для класса OrderResourceImpl.

Добавлен тест для getAllCustomers() и gatAllOrders().

IgushkinV committed:
Homework 18.

Переименование файла.

IgushkinV committed:
Homework 18.

Добавлены тесты для класса SupplierResourceImpl.

Добавлены тесты для класса ProductResourceImpl.

Поставил 0.6 Coverage ratio - jacoco не видит покрытия тестами 3 ресурсов из 4 (по 0% показывает во всех, проме CustomerResourceImpl), хотя в  IDEA там 100% покрытие.
